### ISSUE TITLE
Add early stopping to model training

### ISSUE THREAD

**Issue by larkz** [zalandoresearch/pytorch-ts#45]
Currently the loss function in model training only measures training loss per epoch. Add a feature to include validation loss as an early stopping criteria.
